src/thing/WiFiManager.cpp

```diff
@@ -0,0 +1,69 @@
+#include "thing/WiFiManager.h"
+
+#include <ESP8266WiFi.h>
+#include <DNSServer.h>
+
+
+namespace thing {
+namespace {
+
+const short kDnsPort = 53;
+
+}  // namespace
+
+WiFiManager::WiFiManager() : debug_([](const char*){}) {}
+
+bool WiFiManager::StartAP() {
+  // Use arduino string, casting an int to std::string with arduino C++ support
+  // isn't easy.
+  String ssid("FireThing-");
```

Comment:
Best I can tell there is no easy printf support in arduino. We can use c printf but you need to allocate a buffer and do the manipulations there which is what I was avoiding by just using arduino's string and concatenation.